From 1b7bbc2354d47c1e31213671d8db741f58008556 Mon Sep 17 00:00:00 2001
From: Xing Zheng <zhengxing@rock-chips.com>
Date: Wed, 29 Aug 2018 11:29:57 +0800
Subject: [PATCH] ASoC: rk3308_codec: Fix the dummy loopback

If we playback before capture, the loopback will be
switch to BIST SINE mode by other ADC grps. Let's
fix it.

Change-Id: Ib18a32d87dfed4343edc439bd5c705295eca06f3
Signed-off-by: Xing Zheng <zhengxing@rock-chips.com>
---
 sound/soc/codecs/rk3308_codec.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sound/soc/codecs/rk3308_codec.c b/sound/soc/codecs/rk3308_codec.c
index 86bd9caac018..15b85d1d5dc1 100644
--- a/sound/soc/codecs/rk3308_codec.c
+++ b/sound/soc/codecs/rk3308_codec.c
@@ -2639,6 +2639,7 @@ static int rk3308_codec_open_capture(struct rk3308_codec_priv *rk3308)
 	} else {
 		for (idx = 0; adc_for_each_grp(rk3308, type, idx, &grp); idx++) {
 			if (has_loopback(rk3308->loopback_grp) &&
+			    idx == rk3308->loopback_grp &&
 			    grp == ADC_GRP_SKIP_MAGIC) {
 				/*
 				 * Switch to dummy BIST mode (BIST keep reset
-- 
2.35.3

